### PR TITLE
Integrate login on mobile

### PR DESCRIPTION
**Pull request by nqhhdev**

### CODE REVIEW

**Comment by hoangdat** [comment]
@tk-nguyen I want to change `config.json` for CICD, how can I do?

**Review by hoangdat — APPROVED** [review]

**Comment by tk-nguyen** [comment]
> @tk-nguyen I want to change `config.json` for CICD, how can I do?

Change to what?

File: lib/config/app_config.dart [hunk 1]
```diff
@@ -21,6 +21,10 @@ abstract class AppConfig {
   static double bubbleSizeFactor = 1;
   static double fontSizeFactor = 1;
 
+  static String registrationUrl = 'https://register.tom-dev.xyz/';
+
+  static String twakeWorkplaceHomeserver = 'https://matrix.tom-dev.xyz/';
```
Comment (sherlockvn) [inline lib/config/app_config.dart:26]: why not use const

Comment (hoangdat) [reply lib/config/app_config.dart:26]: we should change it if config.json have other value

File: config.sample.json [hunk 2]
```diff
@@ -6,5 +6,7 @@
   "render_html": true,
   "hide_redacted_events": false,
   "hide_unknown_events": false,
-  "issue_id": ""
+  "issue_id": "",
+  "registration_url": "https://register.tom-dev.xyz/",
+  "twake_workplace_homeserver": "https://matrix.tom-dev.xyz/"
```
Comment (guimard) [inline config.sample.json:11]: Please don't share our platform addresses on a public git. Use example.com

**Review by sherlockvn — APPROVED** [review]

**Comment by hoangdat** [comment]
@tk-nguyen please help us to update some added value in secret

**Comment by tk-nguyen** [comment]
> @tk-nguyen please help us to update some added value in secret

If we're using it in `config.json`, it's not really a secret, because every client will see it.

**Comment by tk-nguyen** [comment]
Anyways, here's the patch if you want to use secrets:
```diff
diff --git a/.github/workflows/gh-pages.yaml b/.github/workflows/gh-pages.yaml
index 443c460db..66bf25d48 100644
--- a/.github/workflows/gh-pages.yaml
+++ b/.github/workflows/gh-pages.yaml
@@ -51,13 +51,15 @@ jobs:
       - name: Build Web version
         env:
           FOLDER: ${{ github.event.pull_request.number }}
+          REGISTRATION_URL: ${{ secrets.REGISTRATION_URL }}
+          TWAKE_WORKPLACE_HOMESERVER: ${{ secrets.TWAKE_WORKPLACE_HOMESERVER }}
         run: |
           flutter config --enable-web
           flutter clean
           flutter pub get
           flutter pub run build_runner build --delete-conflicting-outputs
           flutter build web --release --verbose --source-maps --base-href="/${GITHUB_REPOSITORY##*/}/$FOLDER/"
-          yq '.issue_id = strenv(FOLDER)' config.sample.json > ./build/web/config.json
+          yq '.issue_id = strenv(FOLDER), .registration_id = strenv(REGISTRATION_URL), .twake_workplace_homeserver = strenv(TWAKE_WORKPLACE_HOMESERVER)' config.sample.json > ./build/web/config.json
 
       - name: Configure environments
         id: configure
```

**Pull request merged**